minimint-api/src/db/rocksdb_impl.rs

```diff
@@ -25,6 +25,7 @@ impl Database for rocksdb::OptimisticTransactionDB {
         let prefix = key_prefix.to_vec();
         Box::new(
             self.prefix_iterator(prefix.clone())
+                .filter_map(|res| res.ok())
```

Comment:
we should at least log if we can't read here. Maybe failing is even more appropriate, we shouldn't just ignore errors. What are the circumstances under which this might happen? Are they even recoverable?

Reply:
@Mo0nbase Relevant article published today: https://blog.burntsushi.net/unwrap/

Reply:
Ultimately I think that this is just a bunch of get requests and if some or all of them fail we should essentially handle them the same as a get request if they fail with a exception. Unfortunately get requests shouldn't really fail and if they do it generally indicates something catastrophic has happened with the db. Sometimes the options can be read as null indicating a mis-compiled or mis-loaded rocksdb library which again is probably an issue that a failure would be appropriate. Otherwise all the requests should fail not just one or two so crashing here somewhat makes sense.

Reply:
I think `find_by_prefix` should be infallible. If can return an empty iterator, but otherwise it failing indicates the host lost access to its hard drive (and we should just shut down/crash).